.github/workflows/unit-testing-and-coverage.yml

```diff
@@ -0,0 +1,53 @@
+name: aiosmtpd CI
+
+on:
+  # This is for direct commit to master
+  push:
+    branches: [ master ]
+  # This is for PRs
+  pull_request:
+    branches: [ master ]
+  # Manual/on-demand
+  workflow_dispatch:
+  release:
+    types: [ created, edited, published, prereleased, released ]
+
+jobs:
+  test:
+    strategy:
+      # If a matrix fail, do NOT stop other matrix, let them run to completion
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest ]
+        platform: [ "linux" ]
+        python-version: [ 3.6, 3.7, 3.8, 3.9, pypy3 ]
+        include:
+          - os: windows-latest
+            platform: "mswin"
+            # Always use latest stable branch as indicated on
+            # https://devguide.python.org/#branchstatus
+            # ('stable' means Status == "security")
+            python-version: 3.7
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install tox
+    - name: Execute tox
+      env:
+        PLATFORM: ${{ matrix.platform }}
+      # The "&&" syntax is supported on PowerShell and cmd
+      run: |
+        tox -e "qa,docs,py-cov"
+    - name: Report to codecov
+      # Ubuntu 18.04 came out of the box with 3.6, and LOTS of system are still running
+      # 18.04 happily, so we choose this as the 'canonical' code coverage testing.
+      # One day we'll have to revisit this and bump the version ...
+      if: always() && matrix.python-version == '3.6' && matrix.os == 'ubuntu-latest'
+      uses: codecov/codecov-action@v1
```

Comment:
As suggested by @asvetlov in https://github.com/aio-libs/aiosmtpd/pull/211#discussion_r528799855